ServiceProviderBot/Bot/Dialogs/DialogBase.cs

```diff
@@ -70,14 +72,84 @@ public async Task<DialogTurnResult> ContinueDialogAsync(DialogContext context, C
             return await context.ContinueDialogAsync(cancellationToken);
         }
 
+        protected WaterfallStep[] GenerateUpdateSteps<T>(string serviceName, string totalPropertyName, string openPropertyName,
```

Comment:
Each of the update modules uses the same steps (prompt for open, check if less than total, prompt for waitlist if necessary), so they are now abstracted here to reduce overhead of each module